syntax = "proto3";
package template;




import "collector_center_sdk/model/collector_center/script.proto";
import "collector_center_sdk/model/collector_center/target_range.proto";
import "google/protobuf/struct.proto";

/*
UpdateConfigurationTemplate请求
*/
message UpdateConfigurationTemplateRequest {
    /*
    ID
    */
    string id = 1 ;
    /*
    名称
    */
    string name = 2 ;
    /*
    参数
    */
    google.protobuf.Value kwargs = 3 ;
    /*
    超时时间
    */
    int32 timeout = 4 ;
    /*
    环境变量
    */
    google.protobuf.Value env = 5 ;
    /*
    是否禁用
    */
    bool disabled = 6 ;
    /*
    标签
    */
    repeated string labels = 7 ;
    /*
    脚本
    */
    collector_center.Script script = 8 ;
    /*
    是否忽略不符合要求数据
    */
    bool ignoreInvalid = 9 ;
    /*
    目标动态范围
    */
    collector_center.TargetRange targetRange = 10 ;
    /*
    采集间隔
    */
    int32 interval = 11 ;
    /*
    缓存过期次数（时间）
    */
    int32 cacheTtl = 12 ;
    /*
    执行时间范围（24小时）
    */
    string timeRange = 13 ;
    /*
    采集类型名称
    */
    string clazzName = 14 ;
}

/*
UpdateConfigurationTemplate返回
*/
message UpdateConfigurationTemplateResponse {
    /*
    ID
    */
    string id = 1 ;
}

/*
UpdateConfigurationTemplateApi返回
*/
message UpdateConfigurationTemplateResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    template.UpdateConfigurationTemplateResponse data = 4 ;
}

